### DISCUSSION TITLE
Still useful in june 2024 ?

### DISCUSSION THREAD

**Discussion by xTremePower** [OpenNeptune3D/OpenNept4une#189] [Q&A]
Hello, 

I just changed 8 of my old printers in my printer farm for 8 Neptune 4 Max,
I love these printers, 

I installed on all the last Elegoo FW V1.2.3.0 from 2024.04.29, it work really well, and now I discovered your custom FW, because I love to modify my printers :p 

Is your FW still better than the last official FW ? 

And I don't found on internet any screens or pictures or videos of the neptune working with your FW, I can't even find in your wiki what happens to the touchscreen interface ? it's remain the same ? it's some default Klipper touchscreen after upgrade ?

I need 1 eMMC to USB Adapter by printer or it's just usefull for the update and after you can unplug it ?

And what are the real benefits other the last official FW ?

Thanks a lot

**Comment by TheMeanCanEHdian** [comment]
If your setup works well why mess with it? OpenNept4une lets you run stock Klipper and has all the benefits that brings with it. The screen support is in beta.

**Comment by linhungsam** [comment]
i installed this custom firmware to my regular N4 and it is significantly quieter in comparition to the stock. Just like other mentioned, it also enables the full Klipper features and you dont have to wait for Elegoo to maintain their firmware, which, IMO, really bad.

**Comment by xTremePower** [comment]
Hello
I just updated one of my n4 max to openeptune... And now I'm sure I will
upgrade the 7 others! It's awesome!

We loose some screen functions. But the quality worth it !

**Comment by writepatrick** [comment]
Hey,
after 3-month of using OpenNeptune I have converted my printer (n4pro) back to stock Firmware 1.1.3.1 from elegoo and got one of my best results ever with this printer.

Why did I do this?
1. I never got a hang of the right print settings with the provided profiles (e.g. stringing, random layer-shifts out of nowhere)
2. the benefits did not outweight the disadvantages in usability (screenfunctions)
3. software updates "break" things out of nowhere and you are newer sure if it is "just you" [https://github.com/OpenNeptune3D/display_connector/issues/19](url)
4. elegoo did improve the initial flaws in the machine to a point where I am happy with the machine

if someone is looking for a good orca profile for the stock firmware, I can recomment this [https://www.printables.com/de/model/609896-elegoo-neptune-4pro-orcaslicer-v200-config-new-firmware](url)

**Reply by halfmanbear** [reply]
Fixed Link: https://www.printables.com/de/model/609896-elegoo-neptune-4pro-orcaslicer-v200-config-new-firmware

I downloaded the Optimised orca profile from that link
The cali-cat model I sliced (below) is ~35% slower print time than the provided PLA OpenNept4une profile.

<img width="1440" alt="Screenshot 2024-09-02 at 11 32 27" src="https://github.com/user-attachments/assets/fa7dd19f-84ca-4752-9740-7a6f46191a13">

<img width="1440" alt="Screenshot 2024-09-02 at 12 10 13" src="https://github.com/user-attachments/assets/087bf417-7742-4de0-b329-6f5111b88d5e">

**Reply by writepatrick** [reply]
Hi @halfmanbear,

I am not trying to convince anyone of anything. This was just my experience.
I could compare these profiles again by printing (not just "bench racing"), but I did start with the provided profile and I am sorry to say that but the quality was bad (even my wife recognized the decline in quality). I had massive stringing and observed some skips (I do not know if this was related to the profile since I changed a lot on the printer at that time). 

I also did some slicing and found the issue you are pointing out:
The system "Elegoo generic PLA"-Filament is the culprit here. If I compare both with my eSUN PLA+ filament preset I get similar results (OpenNeptune slighly slower than the other one). BTW there is something with your travel speed, because I get near identical time with your profile and the generic OpenNeptune filament but my travel time is 2m9s with the 250mm/s (nothing changed in the profile)

As I said or maybe not. I am really impressed with the work you are doing, kudos! I could never acchieve what you have acchieved and you stepped in where Elegoo didn't do their homework properly.

It just didn't work out for me, or Elegoo got their shit together finally with the last firmware...

**Reply by xTremePower** [reply]
> Hey, after 3-month of using OpenNeptune I have converted my printer (n4pro) back to stock Firmware 1.1.3.1 from elegoo and got one of my best results ever with this printer.
> 
> Why did I do this?
> 
> 1. I never got a hang of the right print settings with the provided profiles (e.g. stringing, random layer-shifts out of nowhere)
> 2. the benefits did not outweight the disadvantages in usability (screenfunctions)
> 3. software updates "break" things out of nowhere and you are newer sure if it is "just you" [https://github.com/OpenNeptune3D/display_connector/issues/19](url)
> 4. elegoo did improve the initial flaws in the machine to a point where I am happy with the machine
> 
> if someone is looking for a good orca profile for the stock firmware, I can recomment this [https://www.printables.com/de/model/609896-elegoo-neptune-4pro-orcaslicer-v200-config-new-fir](url)

I finally upgraded my 5 Neptune 4 Max to OpenNeptune and I'm never going back ....

The "faults" you mention are only problems and lack of user knowledge ...

First of all: if a person wants to print, they are supposed to be able to create a good profile on the Slicer, without having to follow a profile provided by others ...
As for updates, no one forces you to update OpenNeptune if everything works perfectly, and I've never had any update problems, the problems only happen if you force an update when there is a Repo problem.

I don't know what function is missing on the screen, because with the OpenNeptune Screen module, you can manage everything with the original screen (well, I personally use a tablet fixed to the wall to manage them all at the same time)

And I'm too used to the manual leveling help of Openneptune, it works very easily, you don't even have to think, just launch it and turn the screws as it tells you.

In addition, it is much easier to add a Timelaps, a USB camera, and many other Klipper plugins that are very useful to me, and this without having to be afraid of Elegoo firmware overlays.

Do as you feel, but for me OpenNeptune has only advantages and no faults... after that yes, for an average user, it is to be avoided if he does not even know how to make a SLicer profile himself, if he does not know how to use SSH and correct the problems himself.

**Reply by writepatrick** [reply]
Hi @xTremePower,
I am unsure if I personally insulted you or something. Or why are you going after me?

> I finally upgraded my 5 Neptune 4 Max to OpenNeptune and I'm never going back ....

This is great and I am glad you feel this way.

> The "faults" you mention are only problems and lack of user knowledge ...

Thats a bold claim. The issue with the "CALIBRATE_Z_PROBE_OFFSET" Screen I mentioned here [https://github.com/OpenNeptune3D/display_connector/issues/19](url) and was initally recorgnized by someone else and was subsequently fixed. How was this a "user error", please elaborate.

> First of all: if a person wants to print, they are supposed to be able to create a good profile on the Slicer, without having to follow a profile provided by others ...

From what you wrote, you are not a typical user owning 7 Neptune 4 printers, since you are probably utilizing them in a print farm or you have a serious "hoarding problem". 
I am not sure everbody owning a 3D-Printer is supporting your argument. I have done my calibration before and know the basics on how to diagnose issues but I am not willing to spend hours and hours refining every detail, when somebody else has walked this path already. That is unnecessary.

> As for updates, no one forces you to update OpenNeptune if everything works perfectly, and I've never had any update problems, the problems only happen if you force an update when there is a Repo problem.

Your theory is basically: "I it ain't broke, don't fix it!"
This strategy does not apply to anything related software and especially anything connected to the internet.
A 3D-Printer is a fully fledged linux computer (Armbian for znp-k1) with all security concern and is also connected to the internet. You should always update software if there is an update available. It is the responsibility of the maintainer to ensure this will not break funktionality. 
What have I done wrong?

> I don't know what function is missing on the screen, because with the OpenNeptune Screen module, you can manage everything with the original screen (well, I personally use a tablet fixed to the wall to manage them all at the same time)

I agree most functions are implemented, and I really enjoyed some additional functionality like -"Klipper Adaptive Meshing & Purging" and "Screw Tilt Adjust". As mentioned above the "CALIBRATE_Z_PROBE_OFFSET" issue was only fixable by web-interface. Without it I could not print and use the printer properly.

> In addition, it is much easier to add a Timelaps, a USB camera, and many other Klipper plugins that are very useful to me, and this without having to be afraid of Elegoo firmware overlays.

I agree 100%

My intension was not to make anybody angry. I just wanted to share my thoughts. You get a lot for your money with the Neptune 4 family and it would be even better if they would ship them with proper software. This is where OpenNeptune comes into play stepping in, but it is not 100% polished.

**Reply by linhungsam** [reply]
> Hi @xTremePower, I am unsure if I personally insulted you or something. Or why are you going after me?
> 
> > I finally upgraded my 5 Neptune 4 Max to OpenNeptune and I'm never going back ....
> 
> This is great and I am glad you feel this way.
> 
> > The "faults" you mention are only problems and lack of user knowledge ...
> 
> Thats a bold claim. The issue with the "CALIBRATE_Z_PROBE_OFFSET" Screen I mentioned here [https://github.com/OpenNeptune3D/display_connector/issues/19](url) and was initally recorgnized by someone else and was subsequently fixed. How was this a "user error", please elaborate.
> 
> > First of all: if a person wants to print, they are supposed to be able to create a good profile on the Slicer, without having to follow a profile provided by others ...
> 
> From what you wrote, you are not a typical user owning 7 Neptune 4 printers, since you are probably utilizing them in a print farm or you have a serious "hoarding problem". I am not sure everbody owning a 3D-Printer is supporting your argument. I have done my calibration before and know the basics on how to diagnose issues but I am not willing to spend hours and hours refining every detail, when somebody else has walked this path already. That is unnecessary.
> 
> > As for updates, no one forces you to update OpenNeptune if everything works perfectly, and I've never had any update problems, the problems only happen if you force an update when there is a Repo problem.
> 
> Your theory is basically: "I it ain't broke, don't fix it!" This strategy does not apply to anything related software and especially anything connected to the internet. A 3D-Printer is a fully fledged linux computer (Armbian for znp-k1) with all security concern and is also connected to the internet. You should always update software if there is an update available. It is the responsibility of the maintainer to ensure this will not break funktionality. What have I done wrong?
> 
> > I don't know what function is missing on the screen, because with the OpenNeptune Screen module, you can manage everything with the original screen (well, I personally use a tablet fixed to the wall to manage them all at the same time)
> > I agree most functions are implemented, and I really enjoyed some additional functionality like -"Klipper Adaptive Meshing & Purging" and "Screw Tilt Adjust". As mentioned above the "CALIBRATE_Z_PROBE_OFFSET" issue was only fixable by web-interface. Without it I could not print and use the printer properly.
> 
> > In addition, it is much easier to add a Timelaps, a USB camera, and many other Klipper plugins that are very useful to me, and this without having to be afraid of Elegoo firmware overlays.
> 
> I agree 100%
> 
> My intension was not to make anybody angry. I just wanted to share my thoughts. You get a lot for your money with the Neptune 4 family and it would be even better if they would ship them with proper software. This is where OpenNeptune comes into play stepping in, but it is not 100% polished.

All feedbacks/comments are welcome, that how we improve. The display connector is still in BETA, therefore, there are unexpected bugs. One day, it stopped working and I had to reinstall it and it is back to normal now. 

Expecting major upgrade release from this team.
